### PR TITLE
616 map load performance and much more

### PR DESCRIPTION
**Pull request by tibetsprague**
Faster load time for the map data, load less data for every post and group on the map so we can load more of them from the server more quickly. Then load the extra data needed for posts in in the drawer separately.
This means drawer posts are now paginated and reload from the server anytime a parameter or filter changes on the front-end
https://github.com/Hylozoic/hylo-evo/issues/616

Show members and groups in the drawer in separate tab from posts, alphabetically ordered for now
Fixes https://github.com/Hylozoic/hylo-evo/issues/669

Don't show fulfilled posts on the map
Add Groups to the feature toggles so you can can turn them on or off
You can now sort posts by post date alongside latest activity and popular
Add ability to replace query results cache with new data instead of appending to it 
Add longer delay/debounce to loading map content when changing the viewport
Wait 800ms instead of 300ms so it doesn't constantly try to load new content as you scroll around or zoom in and out. This creates a smoother experience and a lot less load on the servers.
Stop passing lastPostId to saved searches since back-end figures this out now
Export GroupCard from GroupWidget to use in map drawer
Reorder post types to be discussion, request, offer, resource, project, event (members and groups after)
In create modal and map features dialog

Sync map state to URL so you can copy, paste and bookmark
Keep current map view and filters if you browse to other view in the same group and then come back.
Fixes https://github.com/Hylozoic/hylo-evo/issues/606

Don't show toggle for Members on the public context

Fix bug that prevented searching for and jumping to a point location (like an address) (fixes: https://github.com/Hylozoic/hylo-evo/issues/1093)

### CODE REVIEW

File: src/components/Widget/GroupsWidget/GroupsWidget.js [hunk 1]
```diff
@@ -72,3 +54,26 @@ export default class GroupsWidget extends Component {
     )
   }
 }
+
+export function GroupCard ({ group, routeParams, className }) {
```
Comment (thomasgwatson) [inline src/components/Widget/GroupsWidget/GroupsWidget.js:58]: Ehh, Aaron said this version of the groupcard will probably be overwritten once we settle on the new version from the Group Explorer :(

Comment (thomasgwatson) [reply src/components/Widget/GroupsWidget/GroupsWidget.js:58]: Does this version now show up in the context of the map?

**Comment by tibetsprague** [comment]
Yeah thats what I decided to use in the map drawer for now (at Aaron’s suggestion)

File: src/routes/MapExplorer/MapDrawer/MapDrawer.js [hunk 2]
```diff
@@ -43,121 +52,165 @@ function MapDrawer (props) {
...
+            routeParams={routeParams}
+            post={p}
+            styleName='contentCard'
+            constrained
+            expanded={false}
+            key={p.id}
+          />)}
+        </div>
+
+        <ScrollListener onBottom={() => fetchPostsForDrawer(numFetchedPosts, false)} elementId='mapDrawerWrapper' />
```
Comment (thomasgwatson) [inline src/routes/MapExplorer/MapDrawer/MapDrawer.js:157]: Does this one also fetch groups then now?

Comment (thomasgwatson) [reply src/routes/MapExplorer/MapDrawer/MapDrawer.js:157]: or wait, I think you send you just request all public groups

**Review by thomasgwatson — APPROVED** [review]
lgtm

File: src/routes/MapExplorer/MapDrawer/MapDrawer.js (hunk 2 above)
Comment (tibetsprague) [reply src/routes/MapExplorer/MapDrawer/MapDrawer.js:157]: fetchGroups is its own query https://github.com/Hylozoic/hylo-evo/blob/616-map-load-performance/src/routes/MapExplorer/MapExplorer.store.js

yeah i just fetch all public groups in a bounding box for now

File: src/components/Widget/GroupsWidget/GroupsWidget.js (hunk 1 above)
Comment (tibetsprague) [reply src/components/Widget/GroupsWidget/GroupsWidget.js:58]: Yeah thats what I decided to use in the map drawer for now (at Aaron’s suggestion)

**Pull request merged**